service/history/queue/transfer_queue_processor_base.go

```diff
@@ -259,22 +316,14 @@ processorPumpLoop:
 					))
 				}
 				// re-enqueue the event to see if we need keep re-dispatching or load new tasks from persistence
-				t.nextPollTimer.Update(time.Time{})
-				continue processorPumpLoop
-			}
-
-			levels := make(map[int]struct{})
-			now := t.shard.GetTimeSource().Now()
-			for level, pollTime := range t.nextPollTime {
-				if !now.Before(pollTime.time) {
-					levels[level] = struct{}{}
-					delete(t.nextPollTime, level)
-				} else {
-					t.nextPollTimer.Update(pollTime.time)
+				select {
+				case t.processCh <- struct{}{}:
+				default:
 				}
+				continue processorPumpLoop
```

Comment:
Any reason to send signal and continue? Could it just do t.processQueueCollections()?

Reply:
The purpose here is to block all task processing when there are too many pending tasks in the redispatcher (which means we are buffering too many tasks and may run oom), so we can't just do `t.processQueueCollections()`. Continue the processorLoop here so that the check will be executed and backoff again if needed.